library/java/src/io/envoyproxy/envoymobile/engine/EnvoyHTTPStream.java

```diff
@@ -9,14 +9,22 @@
 import java.util.Map;
 
 public class EnvoyHTTPStream {
-
   private final long streamHandle;
   private final JvmCallbackContext callbacksContext;
 
-  EnvoyHTTPStream(long streamHandle, EnvoyHTTPCallbacks callbacks) {
+  /**
+   * Start the stream via the JNI library.
+   */
+  void start() { JniLibrary.startStream(streamHandle, callbacksContext); }
```

Comment:
This change was necessary in order to ensure that mock subclasses of `EnvoyHTTPStream` don't end up calling JNI at all (doing so caused tests to crash). Instead, the initializer now no longer has the side effect of calling the JNI, and the engine starts the stream via this method (similarly to the ObjC engine [here](https://github.com/lyft/envoy-mobile/blob/99fdb261eb614685825c32ab71d4e98f4e579d56/library/objective-c/EnvoyEngineImpl.m#L59-L62))